File: Bedu/Java/Clase3/SharePreFrame/app/src/main/java/com/charlye934/sharepreferencestutorial/Main2Activity.java
package com.charlye934.sharepreferencestutorial;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class Main2Activity extends AppCompatActivity {

    SharedPreferences sharedPreferences;
    TextView tvBienvenida;
    TextView btnBorrra;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        sharedPreferences = getApplicationContext().getSharedPreferences("archivo", Context.MODE_PRIVATE);
        String nombre = sharedPreferences.getString("nombre","");

        tvBienvenida = (TextView)findViewById(R.id.tvBienvenida);
        tvBienvenida.setText("Bienvenido " + nombre);

        btnBorrra = (Button)findViewById(R.id.btnBorrar);
        btnBorrra.setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        borrar();
                    }
                }
        );
    }

    void borrar(){
        sharedPreferences.edit().clear().commit();
        Intent intent = new Intent(getApplicationContext(),MainActivity.class);
        startActivity(intent);
        Toast.makeText(getApplicationContext(),"Se han borrado los datos",Toast.LENGTH_SHORT).show();
    }
}

File: Bedu/kotlin/Clase1-3/SharePreFrame/app/src/main/java/com/charlye934/sharepreferencestutorialkotlin/Main2Activity.kt
package com.charlye934.sharepreferencestutorialkotlin

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main2.*

class Main2Activity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        var sharedPreferences = applicationContext.getSharedPreferences("archivo",Context.MODE_PRIVATE)
        var nombre = sharedPreferences.getString("nombre","")

        tvBienvenida.text = "Bienvenido $nombre"
    }

    fun borrar(view:View){
        var sharedPreferences:SharedPreferences = applicationContext.getSharedPreferences("archivo",Context.MODE_PRIVATE)
        sharedPreferences.edit().clear().commit()
        intent = Intent(applicationContext,MainActivity::class.java)
        startActivity(intent)
        Toast.makeText(applicationContext,"Se han borrado los datos",Toast.LENGTH_SHORT).show()
    }
}
